bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlTreeBuilder.java

```diff
@@ -121,8 +128,11 @@ public BXml parse() {
                         readComment(xmlStreamReader);
                         break;
                     case CDATA:
+                        readCData(xmlStreamReader);
```

Comment:
@prakanth97 Please note that this fix, We need to fix this in xmldata module too.